### PR TITLE
Bugfix FXIOS-8619 [v125] Undoing a bookmark removal re-adds incorrect bookmark

### PR DESCRIPTION
**Pull request by dicarobinho**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8619)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/19126)

## :bulb: Description
Changed URL and page title when undoing a bookmark removal

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+TabToolbarDelegate.swift [hunk 1]
```diff
@@ -318,7 +318,7 @@ extension BrowserViewController: ToolBarActionMenuDelegate {
         presentWithModalDismissIfNeeded(viewController, animated: true)
     }
 
-    func showToast(message: String, toastAction: MenuButtonToastAction) {
+    func showToast(_ bookmarkURL: URL? = nil, _ title: String?, message: String, toastAction: MenuButtonToastAction) {
         switch toastAction {
         case .removeBookmark:
             let viewModel = ButtonToastViewModel(labelText: message,
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+TabToolbarDelegate.swift:324]:
```suggestion
        case .removeBookmark:
            guard let bookmarkURLString = bookmarkURL?.absoluteString else { return }

            let viewModel = ButtonToastViewModel(labelText: message,
```


You could add this and remove some of the optional checks further down (I think if we can't get the url being removed here, no need to run the rest of the code)

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+TabToolbarDelegate.swift [hunk 2]
```diff
@@ -328,8 +328,8 @@ extension BrowserViewController: ToolBarActionMenuDelegate {
                                     theme: themeManager.currentTheme) { [weak self] isButtonTapped in
                 guard let strongSelf = self, let currentTab = strongSelf.tabManager.selectedTab else { return }
                 isButtonTapped ? strongSelf.addBookmark(
-                    url: currentTab.url?.absoluteString ?? "",
-                    title: currentTab.title
+                    url: bookmarkURL?.absoluteString ?? currentTab.url?.absoluteString ?? "",
+                    title: title ?? currentTab.title
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+TabToolbarDelegate.swift:332]:
```suggestion
                    url: bookmarkURLString,
                    title: title ?? currentTab.title
```

**Comment by dataports** [comment]
@dicarobinho Agh, I'm so sorry, I completely missed that this needed reviewing last week. Would you be able to resolve the conflicts and then I can run the tests and merge? 

I added some suggestions for code improvements as well.

File: firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+TabToolbarDelegate.swift [hunk 3]
```diff
@@ -328,8 +328,8 @@ extension BrowserViewController: ToolBarActionMenuDelegate {
                                     theme: themeManager.currentTheme) { [weak self] isButtonTapped in
                 guard let strongSelf = self, let currentTab = strongSelf.tabManager.selectedTab else { return }
```
Comment (dataports) [inline firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+TabToolbarDelegate.swift:329]:
```suggestion
                guard let self, let currentTab = tabManager.selectedTab else { return }
```

If you do this then you can remove `strongSelf` in this closure . I know it's not your change, I know, but figured might as well clean it up :)

**Comment by dicarobinho** [comment]
Closed this because a lot of changes from main.

Created a new child PR: [19312](https://github.com/mozilla-mobile/firefox-ios/pull/19312)

**Pull request closed without merging**